### ISSUE TITLE
Disabling and re-enabling a package causes its keybindings to override custom keybindings

### ISSUE THREAD

**Issue by artoale** [atom/atom-keymap#100]
My custom keybindings stopped working and I suspect this was due to an update to the `vim-mode` package. My keymap has this (besides other stuff):
```cson
'atom-text-editor.vim-mode:not(.insert-mode)':
  'space': 'find-and-replace:show'
  'n': 'find-and-replace:find-next'
  'shift-n': 'find-and-replace:find-previous'
```
But the space-to-find binding stopped working, and the default `vim-mode` one won instead.

I had to just open the keymap.cson file and save it to get it to work again, no big deal, but there should be a way to fix this right? :)

**Comment by Kapaacius** [comment]
+1;

I have rebound: 
'atom-text-editor':
  'ctrl-shift-W': 'editor:select-word'

to

'atom-text-editor':
  'ctrl-w': 'editor:select-word'

But neither of the work after updating to Atom 1.1

**Comment by nathansobo** [comment]
To be clear @artoale, you're saying you lose your bindings after you upgrade a package?

**Comment by maxbrunsfeld** [comment]
@Kapaacius do you have vim-mode installed?

**Comment by Kapaacius** [comment]
@maxbrunsfeld no I do not.

**Comment by maxbrunsfeld** [comment]
@Kapaacius could you report what the keybinding shows when you type `ctrl-w`? Also, could you share your whole `keymap.cson`?

**Comment by maxbrunsfeld** [comment]
It's sounding like these are two separate issues, but I think it's ok to keep them both here for now.

**Comment by Kapaacius** [comment]
https://gist.github.com/b0423443c0239330a82d

This is my current keymap, what used to work.

If I use cmd+w at the moment, it closes my tab (default option). It seems that my custom keymap.cson wont override default values.

**Comment by maxbrunsfeld** [comment]
@Kapaacius I can't reproduce this on my mac; I can override `cmd-w` as expected. I see that you bound `cmd-w` on `atom-text-editor`. Is it possible that you don't have a text editor focused when you type `cmd-w`?

**Comment by Kapaacius** [comment]
I do have it focused...I'll try reinstalling Atom..something I was trying to avoid initially...maybe that helps.

**Comment by artoale** [comment]
@nathansobo I'm unclear whether the update of `vim-mode` is what actually caused the problem or not, but I did noticed an update of it sometimes before noticing the issue, and, since the "winning" binding was `vim-mode`'s one, I thought that might have been the reason for it

**Comment by Kapaacius** [comment]
I actually did get my keymap working again ( reinstall atom helped there )  with 1 exception and that is that 

'atom-text-editor':
  'ctrl-w': 'editor:select-word'

That one just does not want to work properly.

**Comment by Kapaacius** [comment]
I did get my keymap working again...Reinstalling atom and after that some bindings needed to be redefined..but everything seems to be working now.

**Comment by maxbrunsfeld** [comment]
Ok, great. @artoale If you experience your problem again and are able to find some concrete steps to reproduce, please let us know. I'm going to close this one out for right now. Thanks!

**Comment by artoale** [comment]
@maxbrunsfeld Yeah, I did found a way - just disabling and re-enabling `vim-mode` broke mi custom binding (had to re-save the .cson file)

**Comment by maxbrunsfeld** [comment]
Ok, I can reproduce this. Thanks for the report.

**Comment by artoale** [comment]
no problem. If I can help with anything else, please let me know :)

**Comment by maxbrunsfeld** [comment]
@artoale This will be fixed in Atom v1.4.0 and v1.4.0-beta0.

**Comment by artoale** [comment]
Thanks! Awesome!
